### ISSUE TITLE
Google permission change without warning?

### ISSUE THREAD

**Issue by dandotso** [dwyl/web-form-to-google-sheet#26]
I recently came back to test the code which was originally working well. I found people that had not been entered and when I tested the url to call my app gave me a screen "You do not have access to perform that action. Please ask the owner of this item to grant access to you."

I hadn't changed anything since it was last working. I went back into app and deployed it again, with the permissions as in your guide and now it's working again.

The only thing had done was connect it to zapier to move data into a CRM. At one point google blocked this as they thought is looked suspicious, but then it continued fine.

My question is, do you think without warning Google could have changed the permissions of my app to stop it working?

**Comment by mckennapsean** [comment]
That I am unsure of. I don't completely know Zapier, but what exactly was moved? The form? The javascript? The Google Spreadsheet data?

The error message you got sounds just like one if the permissions aren't set right.

I know that you can access your Google Account permissions at the following link, and if you ever cleared or changed these at some point that could affect it. Or if you transferred ownership of a file to someone else (even an organization entity) that may affect permissions...

https://myaccount.google.com/permissions

**Comment by dandotso** [comment]
Thanks for the quick reply. Nothing moved, just my permissions changed for the app without me doing it. I just wanted to know if you had seen this in your experience. 

I did give someone else access for testing... that may have done it then?

**Comment by mckennapsean** [comment]
Gotcha. Personally, I have never seen this happen in my work.

When you gave someone else access to test, what were your steps so we could try to reproduce? That is, did you just add them via the "share" menu for the Google Spreadsheet? What permissions did you grant them? Was it just one person? Did you change the overall level of permission access for the document? (e.g. "anyone with link" to "private, shared only with ...")

**Comment by dandotso** [comment]
Yes I just used the share menu. Full access to they could add a feature where if the email was found in the email column, it wouldn't add a new row, it would update the other columns in the row. This was completed and working with a couple real life people coming through. Then nothing for weeks - I wasn't pushing the site so wasn't expected anything. Then I came back and noticed two new users that hadn't made it into the sheet (they are stored elsewhere too). What happened in that period I don't know, but I think I'll setup a daily call to test access maybe, just to see if it happens again and feedback here as to what happened before access changed.

**Comment by mckennapsean** [comment]
Gotcha. Thanks for the update! This collaborator made changes to the Google Script, right? Do you know if they went through all the steps to redeploy the updated script? (that is, saved as a new version, redeployed the app, and maybe used the appropriate permissions? if any of those weren't done or permissions accidentally changed, then it would simply be human error) I have also seen a few issues from folks posting about Google's API going down in some countries resulting in their inability to send/receive data at all.

I hope it doesn't happen again, but feel free to post anything else you find (as I am sure others will care about that too!). Might be worth having someone try to replicate the issue:
1. setup a new form, spreadsheet, and script
1. add a collaborator via the share menu
1. have that collaborator make changes and redeploy the script
1. check to see if the script has the right permissions, if test submissions make it to the spreadsheet, etc.

**Comment by dandotso** [comment]
I assumed they had done since I got public entries after the work was complete. I'll keep an eye out and report back if it happens again.
